ui/src/app/workflows/components/workflow-logs-viewer/workflow-logs-viewer.tsx

```diff
@@ -60,19 +61,19 @@ export class WorkflowLogsViewer extends React.Component<WorkflowLogsViewerProps,
                         </p>
                     )}
                     {!this.state.error && this.podHasNoLogs() && <p>Pod did not output any logs.</p>}
-                    {this.state.lines.length > 0 && (
+                    {this.state.lineCount > 0 && (
                         <div className='log-box'>
                             <FullHeightLogsViewer
                                 source={{
                                     key: `${this.props.workflow.metadata.name}-${this.props.container}`,
-                                    loadLogs: () => Observable.from(this.state.lines),
+                                    loadLogs: () => this.state.logsObservable,
```

Comment:
Just noting that this component is actually relatively confusing.  For example, you can change the `loadLogs` function completely, however the new observable will only be subscribed to in the event that the `key` property changes

https://github.com/argoproj/argo-ui/blob/master/src/components/logs-viewer/logs-viewer.tsx#L28-L30

I think this would be a good candidate for refactoring; why doesn't it simply accept an `observable` via props and manage the subscription based on when that changes?